pkg/reconciler/apiserversource/resources/oidc_rolebinding.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "knative.dev/eventing/pkg/apis/sources/v1"
+)
+
+// MakeOIDCRole will return the role object config for generating the JWT token
+func MakeOIDCRole(source *v1.ApiServerSource) (*rbacv1.Role, error) {
+	roleName := "create-oidc-token"
+
+	if source.Status.Auth.ServiceAccountName == nil {
```

Comment:
not sure if you check this before already, but `source.Status.Auth` can be nil too